source/opt/decompose_initialized_variables_pass.cpp

```diff
@@ -36,17 +36,32 @@ Pass::Status DecomposeInitializedVariablesPass::Process() {
   auto* module = context()->module();
   bool changed = false;
 
-  // TODO(zoddicus): Handle 'Output' variables
-  // TODO(zoddicus): Handle 'Private' variables
+  std::vector<std::tuple<uint32_t, uint32_t>> global_stores;
+  for (auto iter = module->types_values_begin();
+       iter != module->types_values_end(); ++iter) {
+    Instruction* inst = &(*iter);
+    if (!HasInitializer(inst)) continue;
+
+    changed = true;
+    auto var_id = inst->result_id();
+    auto val_id = inst->GetOperand(3).words[0];
+    global_stores.push_back(std::make_tuple(var_id, val_id));
+    iter->RemoveOperand(3);
```

Comment:
Does this mess up the def-use manager?  You remove the operand which is the use of an id.

Reply:
Yup, fixed it.